libs/milvus/langchain_milvus/vectorstores/milvus.py

```diff
@@ -856,6 +853,82 @@ def add_texts(
 
+        transposed_embeddings = [
+            [embeddings[j][i] for j in range(len(embeddings))]
+            for i in range(len(embeddings[0]))
+        ]
+        # Now:
+        # transposed_embeddings = [
+        #     [f1(a), f2(a)],
+        #     [f1(b), f2(b)],
+        #     [f1(c), f2(c)]
+        # ]
+
+        return self.add_embeddings(
+            texts=texts,
+            embeddings=transposed_embeddings,
+            metadatas=metadatas,
+            timeout=timeout,
+            batch_size=batch_size,
+            ids=ids,
+            **kwargs,
+        )
+
+    def add_embeddings(
+        self,
+        texts: List[str],
+        embeddings: Union[List[List[float]], List[List[List[float]]]],
+        metadatas: Optional[List[dict]] = None,
+        timeout: Optional[float] = None,
+        batch_size: int = 1000,
+        *,
+        ids: Optional[List[str]] = None,
+        **kwargs: Any,
+    ) -> List[str]:
+        """Insert text data with embeddings vectors into Milvus.
+
+        Inserting data when the collection has not be made yet will result
+        in creating a new Collection. The data of the first entity decides
+        the schema of the new collection, the dim is extracted from the first
+        embedding and the columns are decided by the first metadata dict.
+        Metadata keys will need to be present for all inserted values. At
+        the moment there is no None equivalent in Milvus.
+
+        Args:
+            text (List[str]): the texts to insert
+            embeddings (Union[List[List[float]], List[List[List[float]]]]):
+                A vector embeddings for each text (in case of a single vector)
+                or list of vectors for each text (in case of multi-vector)
+            text_embeddings (
+                Iterable[Tuple[str, Union[List[float], List[List[float]]]]]
+            ):
+                The texts with embeddings vectors,
+                it is assumed that they all fit in memory.
+            metadatas (Optional[List[dict]]): Metadata dicts attached to each of
+                the texts. Defaults to None.
+            should be less than 65535 bytes. Required and work when auto_id is False.
+            timeout (Optional[float]): Timeout for each batch insert. Defaults
+                to None.
+            batch_size (int, optional): Batch size to use for insertion.
+                Defaults to 1000.
+            ids (Optional[List[str]]): List of text ids. The length of each item
+
+        Raises:
+            MilvusException: Failure to add texts and embeddings
+
+        Returns:
+            List[str]: The resulting keys for each inserted element.
+        """
+        from pymilvus import Collection, MilvusException
+
+        # Transpose embeddings to make it a list of embeddings of each type.
+        embeddings = [
+            [embeddings[j][i] for j in range(len(embeddings))]
+            for i in range(len(embeddings[0]))
+        ]
```

Comment:
But now I had to "transpose" again to the original view -> which I don't like because it's quiet heavy operation